Source/bannedwords.cpp

```diff
@@ -0,0 +1,83 @@
+// bannedwords.cpp
+
+#include "bannedwords.hpp"
+
+#include "utils/file_util.h"
+#include "utils/log.hpp"
+#include "utils/paths.h"
+
+#include <algorithm>
+#include <cctype>
+
+namespace devilution {
+
+std::vector<std::string> BannedWords;
+
+void LoadBannedWords()
+{
+	BannedWords.clear();
+	const std::string path = paths::ConfigPath() + "bannedwords.txt";
+
+	// Create file if it doesn't exist
+	if (!FileExists(path.c_str())) {
+		if (FILE *file = OpenFile(path.c_str(), "wb")) {
```

Comment:
`FileExists()` does not provide a strong enough guarantee to avoid truncating an existing file with `wb`.

https://github.com/diasurgical/DevilutionX/blob/ecb3e46199708946e4c563a953b0e67c33958f4f/Source/mpq/mpq_writer.cpp#L97-L99